src/util/color/colorpalette.cpp

```diff
@@ -0,0 +1,55 @@
+#include "colorpalette.h"
+
+namespace {
+
+constexpr mixxx::RgbColor kColorMixxxRed(0xC50A08);
+constexpr mixxx::RgbColor kColorMixxxYellow(0x32BE44);
+constexpr mixxx::RgbColor kColorMixxxGreen(0x42D4F4);
+constexpr mixxx::RgbColor kColorMixxxCeleste(0xF8D200);
+constexpr mixxx::RgbColor kColorMixxxBlue(0x0044FF);
+constexpr mixxx::RgbColor kColorMixxxPurple(0xAF00CC);
+constexpr mixxx::RgbColor kColorMixxxPink(0xFCA6D7);
+constexpr mixxx::RgbColor kColorMixxxWhite(0xF2F2FF);
+
+// Replaces "no color" values and is used for new cues if auto_hotcue_colors is
+// disabled
+constexpr mixxx::RgbColor kSchemaMigrationReplacementColor(0xFF8C00);
+
+} // anonymous namespace
+
+const ColorPalette ColorPalette::mixxxHotcuePalette =
+        ColorPalette(
+                QStringLiteral("Mixxx Hotcue Colors"),
+                QList<mixxx::RgbColor>{
+                        kColorMixxxRed,
+                        kColorMixxxYellow,
+                        kColorMixxxGreen,
+                        kColorMixxxCeleste,
+                        kColorMixxxBlue,
+                        kColorMixxxPurple,
+                        kColorMixxxPink,
+                        kColorMixxxWhite,
+                });
+
+const mixxx::RgbColor ColorPalette::kDefaultCueColor = kSchemaMigrationReplacementColor;
+
+mixxx::RgbColor ColorPalette::nextColor(mixxx::RgbColor color) const {
+    //  Return first color if color not in palette ((-1 + 1) % size)
```

Comment:
To mirror previousColor, shouldn't this return the last color, not the first?

Reply:
I'm not sure. The use case is: I assign some color to a hotcue, then switch to another palette that does not contain that color. I press the color next/prev buttons on my controller. For "next" I'd expect to get the first color in the new palette. Maybe I should modify `previousColor` to return the last color instead.

Reply:
https://github.com/mixxxdj/mixxx/pull/2520/commits/714f4467b1d912edd2f9511bfc46507fa9f8d25a

Reply:
I imagine these will be connected to buttons on controllers so it should make sense from a scrolling point of view.  yes, previouscolor might want to return the last color